Documentation/driver-api/virtual_bus.rst

```diff
@@ -0,0 +1,164 @@
+===========
+Virtual Bus
+===========
+
+See <linux/virtual_bus.h> for the models for virtbus_device and virtbus_driver.
+
+In some subsystems, the functionality of the core device (PCI/ACPI/other) may
+be too complex for a single device to be managed as a monolithic block or
+a part of the functionality might need to be exposed to a different subsystem.
+Splitting the functionality into smaller orthogonal devices would make it
+easier to manage data, power management and domain-specific communication with
+the hardware. A key requirement for such a split is that there is no dependency
+on a physical bus, device, register accesses or regmap support. These
+individual devices split from the core cannot live on the platform bus as they
+are not physical devices that are controlled by DT/ACPI. The same argument
+applies for not using MFD in this scenario as it relies on individual function
+devices being physical devices that are DT enumerated.
+
+An example for this kind of requirement is the audio subsystem where a single
+IP handles multiple entities such as HDMI, Soundwire, local devices such as
+mics/speakers etc. The split for the core's functionality can be arbitrary or
+be defined by the DSP firmware topology and include hooks for test/debug. This
+allows for the audio core device to be minimal and tightly coupled with
+handling the hardware-specific logic and communication.
+
+The virtual bus is aimed at being a minimal software bus where virtbus
+devices carved out to represent a part of their core device's functionality can
+live on. This bus is also meant to be generic and be available to be used by
```

Comment:
@ranj063 Complex language detected. How about: The virtual bus is aimed at being a minimal software bus on which virtbus devices, carved out to represent a part of their core device's functionality, can live.

Still pretty complex. Maybe grammarly could do better :)